torch/csrc/jit/runtime/profiling_record.cpp

```diff
@@ -214,6 +233,31 @@ void ProfilingRecord::instrumentBlock(Block* block) {
           (needsProfiledInputs(n) || needsProfiledOutput(i->node()))) {
         insertShapeProfile(n, offset);
       }
+
+      if (i->type()->cast<OptionalType>() && hasGradSumToSizeUses(i)) {
```

Comment:
nitpick: maybe a quick comment that we are profiling a definition vs uses `prim::profile`s profile?